### PR TITLE
Fix assertion errors in debug mode

### PR DESCRIPTION
**Pull request by tkoolen**
Fix assertion errors in debug mode. Decided to pull these out of the HandC pull request to make things clearer.
Related: #992

### CODE REVIEW

File: systems/plants/shapes/Geometry.cpp [hunk 1]
```diff
@@ -40,7 +40,7 @@ namespace DrakeShapes
   void Geometry::getBoundingBoxPoints(double x_half_width, double y_half_width, double z_half_width, Eigen::Matrix3Xd &points) const
   {
     // Return axis-aligned bounding-box vertices
-      points.resize(Eigen::NoChange, NUM_BBOX_POINTS);
+      points.resize(3, NUM_BBOX_POINTS);
```
Comment (tkoolen) [inline systems/plants/shapes/Geometry.cpp]: Just found out that this one actually wasn't necessary.

**Comment by RussTedrake** [comment]
looks good to me.  will just wait for build servers.

**Comment by tkoolen** [comment]
You can either pull this, then pull #986, or only pull #986.

**Comment by tkoolen** [comment]
I don't see how these changes could have caused this test failure.

**Comment by RussTedrake** [comment]
that's almost certainly not you.  it's been an issue before (#330), so I've just re-opened it.

**Pull request merged**